src/data/chats/mod.rs

```diff
@@ -5,43 +5,75 @@ pub mod model_loader;
 use anyhow::{Context, Result};
 use chat::{Chat, ChatEntityAction, ChatID};
 use chat_entity::ChatEntityId;
-use makepad_widgets::ActionTrait;
+use makepad_widgets::{error, ActionDefaultRef, ActionTrait, Cx, DefaultNone};
 use model_loader::ModelLoader;
 use moly_backend::Backend;
-use moly_mofa::{MofaAgent, MofaBackend};
+use moly_mofa::{AgentId, MofaAgent, MofaClient, MofaServerId, MofaServerResponse};
 use moly_protocol::data::*;
 use moly_protocol::protocol::Command;
+use std::collections::HashMap;
 use std::fs;
-use std::sync::mpsc::channel;
+use std::sync::mpsc::{self, channel};
 use std::{cell::RefCell, path::PathBuf, rc::Rc};
 
 use super::filesystem::setup_chats_folder;
 
+#[derive(Clone, Debug)]
+pub struct MofaServer {
+    pub address: String,
+    pub client: Rc<MofaClient>,
+    pub connection_status: MofaServerConnectionStatus,
+}
+
+impl MofaServer {
+    pub fn is_local(&self) -> bool {
+        self.address.starts_with("http://localhost")
+    }
+}
+
+/// The connection status of the server
+#[derive(Debug, Clone, PartialEq)]
+pub enum MofaServerConnectionStatus {
+    Connecting,
+    Connected,
+    Disconnected,
+}
+
+#[derive(Debug, DefaultNone, Clone)]
+pub enum MoFaTestServerAction {
+    Success(String, Vec<MofaAgent>),
+    Failure(Option<String>),
+    None,
+}
+
 pub struct Chats {
     pub backend: Rc<Backend>,
-    pub mae_backend: Rc<MofaBackend>,
     pub saved_chats: Vec<RefCell<Chat>>,
 
     pub loaded_model: Option<File>,
     pub model_loader: ModelLoader,
 
+    pub mofa_servers: HashMap<MofaServerId, MofaServer>,
+    pub available_agents: HashMap<AgentId, MofaAgent>,
+
     current_chat_id: Option<ChatID>,
     chats_dir: PathBuf,
 
     override_port: Option<u16>,
 }
```

Comment:
I don't love that I had to put these here. (also in generally I don't love polluted `mod.rs` files)
Originally it felt more appropriate to put these `HashMaps` in Store and have the related operations there (or in a separate module). But due to how the `ChatPanel` and other parts of the UI interact with `/chats/mod.rs` I had to keep everything together for the sake of sanity and have some consistency.
I think after this PR we can think of some refactoring, I don't think this `Chats` abstraction is the best, and there's some mix up between this and similar things in the `Store`.
Somewhat related we can also add some traits for unifying the MoFa and Models operations. I didn't want to make this PR harder to review.